Formula/hadoop.rb

```diff
@@ -15,27 +15,51 @@ class Hadoop < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "92cb92589ac48035f2482eca9321cebe31dfd3873272da601e24f30cc758cb5e"
   end
 
-  depends_on "openjdk"
+  # WARNING: Check https://cwiki.apache.org/confluence/display/HADOOP/Hadoop+Java+Versions before updating JDK version
+  depends_on "openjdk@11"
 
   conflicts_with "yarn", because: "both install `yarn` binaries"
 
   def install
     rm_f Dir["bin/*.cmd", "sbin/*.cmd", "libexec/*.cmd", "etc/hadoop/*.cmd"]
     libexec.install %w[bin sbin libexec share etc]
     Dir["#{libexec}/bin/*"].each do |path|
-      (bin/File.basename(path)).write_env_script path, JAVA_HOME: Formula["openjdk"].opt_prefix
+      (bin/File.basename(path)).write_env_script path,
+        JAVA_HOME: Formula["openjdk@11"].opt_prefix, HADOOP_LOG_DIR: "#{var}/hadoop"
     end
     Dir["#{libexec}/sbin/*"].each do |path|
-      (sbin/File.basename(path)).write_env_script path, JAVA_HOME: Formula["openjdk"].opt_prefix
+      (sbin/File.basename(path)).write_env_script path,
+        JAVA_HOME: Formula["openjdk@11"].opt_prefix, HADOOP_LOG_DIR: "#{var}/hadoop"
     end
     Dir["#{libexec}/libexec/*.sh"].each do |path|
-      (libexec/File.basename(path)).write_env_script path, JAVA_HOME: Formula["openjdk"].opt_prefix
+      (libexec/File.basename(path)).write_env_script path,
+        JAVA_HOME: Formula["openjdk@11"].opt_prefix, HADOOP_LOG_DIR: "#{var}/hadoop"
     end
     # Temporary fix until https://github.com/Homebrew/brew/pull/4512 is fixed
     chmod 0755, Dir["#{libexec}/*.sh"]
   end
 
   test do
     system bin/"hadoop", "fs", "-ls"
+
+    # Test if resource manager can start successfully
+    pid = Process.spawn({
+      "JAVA_HOME" => Formula["openjdk@11"].opt_prefix,
+      "CLASSPATH" => "#{libexec}/etc/hadoop:" \
+                     "#{libexec}/share/hadoop/common/lib/*:" \
+                     "#{libexec}/share/hadoop/common/*:" \
+                     "#{libexec}/share/hadoop/hdfs:" \
+                     "#{libexec}/share/hadoop/hdfs/lib/*:" \
+                     "#{libexec}/share/hadoop/hdfs/*:" \
+                     "#{libexec}/share/hadoop/mapreduce/*:" \
+                     "#{libexec}/share/hadoop/yarn:" \
+                     "#{libexec}/share/hadoop/yarn/lib/*:" \
+                     "#{libexec}/share/hadoop/yarn/*:" \
+                     "#{libexec}/share/hadoop/yarn/timelineservice/*:" \
+                     "#{libexec}/share/hadoop/yarn/timelineservice/lib/*",
+    }, "#{Formula["openjdk@11"].bin}/java", "org.apache.hadoop.yarn.server.resourcemanager.ResourceManager")
+    sleep 8
+    Process.getpgid pid
+    system "curl", "http://127.0.0.1:8088"
```

Comment:
Can we specify the `8080` so we can select an available port using `free_port`?

Reply:
Don't know what ```free_port``` is.  But changed webapp port to 8080.

Reply:
Try
```
grep -r free_port "$(brew --repo homebrew/core)/Formula"
```
for many examples.

Reply:
done